docs/reference/aggregation.md

```diff
@@ -0,0 +1,488 @@
+}
+```
+
+
+###### **min_doc_count**
+
+The minimum number of documents in a bucket to be returned. Defaults to 0.
+
+###### **hard_bounds**
+
+Limits the data range to [min, max] closed interval.
+This can be used to filter values if they are not in the data range.
+hard_bounds only limits the buckets, to force a range set both extended_bounds and hard_bounds to the same range.
+
+```json
+{
+    "prices": {
+        "histogram": {
+            "field": "price",
+            "interval": 10,
+            "hard_bounds": {
+                "min": 0,
+                "max": 100
+            }
+        }
+    }
+}
+```
+
+###### **extended_bounds**
+
+Can be set to extend your bounds. The range of the buckets is by default defined by the data range of the values of the documents. As the name suggests, this can only be used to extend the value range. If the bounds for min or max are not extending the range, the value has no effect on the returned buckets.
+Cannot be set in conjunction with min_doc_count > 0, since the empty buckets from extended bounds would not be returned.
+
+```json
+{
+    "prices": {
+        "histogram": {
+            "field": "price",
+            "interval": 10,
+            "extended_bounds": {
+                "min": 0,
+                "max": 100
+            }
+        }
+    }
+}
+```
+
+
+### Range
+
+Provide user-defined buckets to aggregate on. Two special buckets will automatically be created to cover the whole range of values.
+The provided buckets have to be continous. During the aggregation, the values extracted from the fast_field field will be checked against each bucket range.
+Note that this aggregation includes the from value and excludes the to value for each range.
+
+```json
+{
+    "my_ranges": {
+        "field": "score",
+        "ranges": [
+            { "to": 3.0 },
+            { "from": 3.0, "to": 7.0 },
+            { "from": 7.0, "to": 20.0 },
+            { "from": 20.0 }
+        ]
+    }
+}
+```
+
+#### Limitations/Compatibility
+
+Overlapping ranges are not yet supported.
+
+The keyed parameter (elasticsearch) is not yet supported.
+
+
+#### Parameters
+
+###### **field**
+
+The field to aggregate on.
+
+###### **ranges**
+
+The list of buckets, with from and to values. 
+The from value is inclusive in the range.
+The to value is not inclusive in the range.
+
+The first bucket can omit the `from` value, and the last bucket the `to` value.
+Note that this aggregation includes the from value and excludes the to value for each range. Extra buckets will be created until the first to, and last from, if necessary.
+
+
+### Terms
+
+Creates a bucket for every unique term.
+
+```json
+{
+    "genres": {
+        "terms":{ "field": "genre" }
+    }
+}
+```
+
+
+#### Terminology
+Shard parameters are supposed to be equivalent to elasticsearch shard parameter.
+Since they are
+
+#### Document count error
+To improve performance, results from one segment are cut off at `segment_size`. On a index with
```

Comment:
I think it's better to talk about split and not segment.

Reply:
I guess `segment_size` is from elasticsearch API, so maybe explain that a quickwit split is equivalent to an elasticsearch segment

Reply:
segment_size is a tantivy specific variable. This is where we deviate a little from elasticsearch since we don't have a global termmapping

Reply:
We also have `shard_size`, which is not used in quickwit currently. There is actually something missing here.
I added an alias for shard_size to split_size (https://github.com/quickwit-oss/tantivy/pull/1358) and this split_size would be set by quickwit to set the shard_size, and we only document the split_size parameter.